src/modules/esl-utils/dom/visible.ts

```diff
@@ -0,0 +1,33 @@
+import {Rect} from './rect';
+import {getListScrollParents} from './scroll';
+import {findClosestBy, findHost} from './traversing';
+import {getWindowRect} from './window';
+
+export interface VisibilityOptions {
+  /** Element will be considered invisible if opacity set to '0' */
+  opacity?: boolean;
+  /** Element will be considered invisible if visibility set to 'hidden' */
+  visibility?: boolean;
+  /** Element will be considered invisible if it's not in viewport */
+  viewport?: boolean;
+}
+
+/**
+ * Check if specified element is visible
+ * @param el - element to be checked
+ * @param options - object of additional visibility options to include
+ */
+export function isVisible(el: HTMLElement, options: VisibilityOptions = {visibility: true}): boolean {
+  if (!el.getClientRects().length) return false;
+  if (options.visibility && findHost(el, (host) => getComputedStyle(host).visibility === 'hidden')) return false;
+  if (options.opacity &&
+    findClosestBy(el, (parent) => parent instanceof Element && getComputedStyle(parent).opacity === '0')) return false;
+  return !(options.viewport && !isInViewport(el));
+}
```

Comment:
Is it correct that `isVisible(el, {})` always returns `true`?

Reply:
Not always. Function will always check for display css property